services/dynamic/dynamic-xml.service.js

```diff
@@ -27,18 +27,7 @@ module.exports = class DynamicXml extends BaseService {
     }
   }
 
-  async handle(namedParams, { url, query: pathExpression, prefix, suffix }) {
-    // e.g. //book[2]/@id
-    const pathIsAttr = (
-      pathExpression.split('/').slice(-1)[0] || ''
-    ).startsWith('@')
-
-    const { buffer } = await this._request({
-      url,
-      options: { headers: { Accept: 'application/xml, text/xml' } },
-      errorMessages,
-    })
-
+  transform({ pathExpression, buffer, pathIsAttr }) {
```

Comment:
I really like to see this as a separate function - it's more readable and easier to test.